src/main/java/net/glowstone/entity/objects/GlowItemFrame.java

```diff
@@ -44,81 +43,21 @@ public GlowItemFrame(GlowPlayer player, Location location, BlockFace clickedface
                 player.setItemInHand(is);
             }
         }
-        metadata.set(MetadataIndex.ITEM_FRAME_ROTATION, 0);
-        metadata.set(MetadataIndex.ITEM_FRAME_ITEM, new ItemStack(Material.AIR));
-
-        itemInFrame = Material.AIR;
-    }
-
-    private static byte getFacingNumber(BlockFace face) {
-        switch (face) {
-            case SOUTH:
-                return 0;
-            case WEST:
-                return 1;
-            case NORTH:
-                return 2;
-            case EAST:
-                return 3;
-            default:
-                return 0;
-        }
-    }
 
-    private static BlockFace getFace(int face) {
-        switch (face) {
-            case 0:
-                return BlockFace.SOUTH;
-            case 1:
-                return BlockFace.WEST;
-            case 2:
-                return BlockFace.NORTH;
-            case 3:
-                return BlockFace.EAST;
-            default:
-                return BlockFace.SOUTH;
-        }
+        setRotation(Rotation.NONE);
+        setItem(new ItemStack(Material.AIR));
     }
 
     // //////////////////////////////////////////////////////////////////////////
     // Overrides
 
-    private static BlockFace inverseGetFace(int face) {
-        switch (face) {
-            case 0:
-                return BlockFace.NORTH;
-            case 1:
-                return BlockFace.EAST;
-            case 2:
-                return BlockFace.SOUTH;
-            case 3:
-                return BlockFace.WEST;
-            default:
-                return BlockFace.NORTH;
-        }
-    }
-
-    public void setItemInFrame(ItemStack is) {
-        if (is == null) {
-            is = new ItemStack(Material.AIR, 1);
-        }
-        is.setAmount(1);
-        itemInFrame = is.getType();
-        metadata.set(MetadataIndex.ITEM_FRAME_ITEM, is);
-        metadata.set(MetadataIndex.ITEM_FRAME_ROTATION, 0);
-    }
-
-    public void setItemFrameRotation(int rotation) {
-        metadata.set(MetadataIndex.ITEM_FRAME_ROTATION, rotation);
-    }
-
     @Override
     public boolean entityInteract(GlowPlayer player, InteractEntityMessage message) {
         if (message.getAction() == Action.INTERACT.ordinal()) {
-            if (itemInFrame == Material.AIR) {
+            if (InventoryUtil.isEmpty(getItem())) {
                 ItemStack isInHand = player.getItemInHand();
```

Comment:
I'll correct myself: it doesn't seem like you can place items from the off-hand for Item Frames and Armor Stands (see: https://bugs.mojang.com/browse/MC-84771). Please add a comment or `todo` around this check in case this vanilla bug is fixed in the future (although it has been opened for nearly 2 years). The check remains OK as long as they don't fix this.